packages/marketplace/src/logic/FetchRemotes.ts

```diff
@@ -0,0 +1,308 @@
+import { BLACKLIST_URL, ITEMS_PER_REQUEST } from "../constants";
+import { CardItem, Snippet } from "../types/marketplace-types";
+import { addToSessionStorage, processAuthors } from "./Utils";
+
+import { RepoTopic } from "../types/marketplace-types";
+import snippetsJSON from "../resources/snippets";
+
+// TODO: add sort type, order, etc?
+// https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-for-repositories#search-by-topic
+// https://docs.github.com/en/rest/reference/search#search-repositories
+
+/**
+ * Query GitHub for all repos with the requested topic
+ * @param tag The tag ("topic") to search for
+ * @param page The query page number
+ * @returns Array of search results (filtered through the blacklist)
+ */
+export async function getTaggedRepos(tag: RepoTopic, page = 1, BLACKLIST: string[] = [], query?: string) {
+	// www is needed or it will block with "cross-origin" error.
+	let url = query
+		? `https://api.github.com/search/repositories?q=${encodeURIComponent(`${query}+topic:${tag}`)}&per_page=${ITEMS_PER_REQUEST}`
+		: `https://api.github.com/search/repositories?q=${encodeURIComponent(`topic:${tag}`)}&per_page=${ITEMS_PER_REQUEST}`;
+
+	// We can test multiple pages with this URL (58 results), as well as broken iamges etc.
+	// let url = `https://api.github.com/search/repositories?q=${encodeURIComponent("topic:spicetify")}`;
+	if (page) url += `&page=${page}`;
+	// Sorting params (not implemented for Marketplace yet)
+	// if (sortConfig.by.match(/top|controversial/) && sortConfig.time) {
+	//     url += `&t=${sortConfig.time}`
+	const allRepos = await fetch(url)
+		.then((res) => res.json())
+		.catch(() => []);
+	if (!allRepos.items) {
+		Spicetify.showNotification("Too Many Requests, Cool Down.");
```

Comment:
This is only one example, but it seems as though Spicetify is no longer a valid object?